### PR TITLE
Indexer: add recipients to address table and address number API

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

Per feedback from Adeniyi on wave 2, we want to include recipients in address table as well

future work: talk with FE team to see how they plan to query address and plan accordingly

## Test Plan 

Added test and CI

### CODE REVIEW

File: crates/sui-indexer/tests/integration_tests.rs [hunk 1]
```diff
@@ -117,7 +93,69 @@ mod pg_integration {
...
                 Some(ExecuteTransactionRequestType::WaitForLocalExecution),
             )
-            .await?;
+            .await
+            .map_err(|e| anyhow::anyhow!(e))?;
+        Ok((tx_response, *sender, *recipient, gas_objects))
+    }
+
+    #[tokio::test]
+    async fn test_genesis_sync() {
```
Comment (gegaowp) [inline crates/sui-indexer/tests/integration_tests.rs:102]: this is a bit misleading, I moved another block of codes to `execute_simple_transfer` actually

File: crates/sui-indexer/src/models/transactions.rs [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 use sui_json_rpc_types::{
     OwnedObjectRef, SuiObjectRef, SuiTransaction, SuiTransactionDataAPI, SuiTransactionEffects,
     SuiTransactionEffectsAPI,
 };
 
+use crate::errors::IndexerError;
+use crate::schema::transactions;
+use crate::schema::transactions::transaction_digest;
+use crate::types::SuiTransactionFullResponse;
+use crate::PgPoolConnection;
```
Comment (wlmyng) [inline crates/sui-indexer/src/models/transactions.rs:16]: question: is this just the result of some formatter? Looks like L4-9 were just moved to L12-16

File: crates/sui-indexer/tests/integration_tests.rs [hunk 3]
```diff
@@ -117,7 +93,68 @@ mod pg_integration {
...
+        drop(handle);
+    }
+
+    #[tokio::test]
+    async fn test_total_address() -> Result<(), anyhow::Error> {
+        let (mut test_cluster, indexer_rpc_client, store, _handle) = start_test_cluster().await;
+        // Allow indexer to sync genesis
+        wait_until_next_checkpoint(&store).await;
+        let (tx_response, _sender, _recipient, _gas_objects) =
+            execute_simple_transfer(&mut test_cluster, &indexer_rpc_client)
```
Comment (wlmyng) [inline crates/sui-indexer/tests/integration_tests.rs:138]: question: mostly for my understanding, is `execute_simple_transfer` needed to test that there are 10 addresses?

File: crates/sui-indexer/src/models/addresses.rs [hunk 4]
```diff
@@ -1,25 +1,14 @@
...
+
 #[derive(Queryable, Insertable, Debug)]
 #[diesel(table_name = addresses, primary_key(account_address))]
 pub struct Address {
     pub account_address: String,
     pub first_appearance_tx: String,
     pub first_appearance_time: i64,
 }
-
-impl From<&Transaction> for Address {
```
Comment (wlmyng) [inline crates/sui-indexer/src/models/addresses.rs:17]: question: so previously we'd derive address from transaction, but now we basically want to lift addresses to be queried directly? Would it still be useful to keep this

**Review by wlmyng — APPROVED** [review]
:ok:

File: crates/sui-indexer/src/models/addresses.rs (hunk 4 above)
Comment (gegaowp) [reply crates/sui-indexer/src/models/addresses.rs:17]: this PR changed the semantic meaning of an address now, before this PR, it was sender or an txn, but after this PR, it is not sender or recipient account address.

txn has a `recipients` field but it is strings of Owner, which can be addresses owners of other types like
```
pub enum Owner {
    /// Object is exclusively owned by a single address, and is mutable.
    AddressOwner(SuiAddress),
    /// Object is exclusively owned by a single object, and is mutable.
    /// The object ID is converted to SuiAddress as SuiAddress is universal.
    ObjectOwner(SuiAddress),
    /// Object is shared, can be used by any address, and is mutable.
    Shared {
        /// The version at which the object became shared
        initial_shared_version: SequenceNumber,
    },
    /// Object is immutable, and hence ownership doesn't matter.
    Immutable,
}
```

File: crates/sui-indexer/tests/integration_tests.rs (hunk 3 above)
Comment (gegaowp) [reply crates/sui-indexer/tests/integration_tests.rs:138]: good call, I wanted to test other things as well but shave that part away.

File: crates/sui-indexer/src/models/transactions.rs (hunk 2 above)
Comment (gegaowp) [reply crates/sui-indexer/src/models/transactions.rs:16]: just re-org the `use`, the idiomatic order is std -> external -> crate, I put sui / Mysten ones between external and crate.

**Pull request merged**